test.js

```diff
@@ -65,23 +65,25 @@ test('match exact URLs', t => {
 });
 
 test('match URLs in text', t => {
-	const fixture = [
-		'Lorem ipsum //dolor.sit',
-		'<a href="http://example.com">example.com</a>',
-		'[and another](https://another.example.com)',
-		'Foo //bar.net/?q=Query with spaces'
-	].join('\n');
+	const fixture = `
+		Lorem ipsum //dolor.sit
+		<a href="http://example.com">example.com</a>
+		<a href="http://example.com/with-path">with path</a>
+		[and another](https://another.example.com) and
```

Comment:
I'm afraid it's not that easy... Any of `()[]` are all valid characters in an URL, e.g: `http://example.com/foo_(bar)?q[]=1&q[]=2`. This module does not have a separate regex part for the query string and the query string can take almost any characters which is why it's so difficult to don't match a possible closing parentheses.